test/functional/mempool_ephemeral_dust.py

```diff
@@ -0,0 +1,493 @@
+        assert_mempool_contents(self, self.nodes[0], expected=[dusty_tx["tx"], sweep_tx_2["tx"]], sync=False)
+
+        # TRUC transactions restriction for ephemeral dust disallows further spends of ancestor chains
+        child_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=sweep_tx_2["new_utxos"], version=3)
+        assert_raises_rpc_error(-26, "TRUC-violation", self.nodes[0].sendrawtransaction, child_tx["hex"])
+
+        self.nodes[0].reconsiderblock(reconsider_block_res["hash"])
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        self.log.info("Test that ephemeral dust tx with fees or multi dust don't enter mempool via reorg")
+        multi_dusty_tx = self.wallet.create_self_transfer_multi(fee_per_output=0, version=3)
+        self.add_output_to_create_multi_result(multi_dusty_tx)
+        self.add_output_to_create_multi_result(multi_dusty_tx)
+
+        block_res = self.nodes[0].rpc.generateblock(self.wallet.get_address(), [multi_dusty_tx["hex"]])
+        self.nodes[0].invalidateblock(block_res["hash"])
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        # With fee and one dust
+        dusty_fee_tx = self.wallet.create_self_transfer_multi(fee_per_output=1, version=3)
+        self.add_output_to_create_multi_result(dusty_fee_tx)
+
+        block_res = self.nodes[0].rpc.generateblock(self.wallet.get_address(), [dusty_fee_tx["hex"]])
+        self.nodes[0].invalidateblock(block_res["hash"])
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        # Re-connect and make sure we have same state still
+        self.connect_nodes(0, 1)
+        self.sync_all()
+
+    def test_acceptnonstd(self):
+        self.log.info("Test that multi-dust and non-0 fee dust is still allowed when acceptnonstdtxn=1")
+        self.restart_node(0, extra_args=["-acceptnonstdtxn=1"])
+        self.restart_node(1, extra_args=["-acceptnonstdtxn=1"])
+        self.connect_nodes(0, 1)
+
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        # Double dust, both unspent, with fees. Would have failed individual checks.
+        dusty_tx = self.wallet.create_self_transfer_multi(fee_per_output=1000, version=3)
+        self.add_output_to_create_multi_result(dusty_tx)
+        self.add_output_to_create_multi_result(dusty_tx)
+        self.nodes[0].rpc.sendrawtransaction(dusty_tx["hex"])
+
+        assert_mempool_contents(self, self.nodes[0], expected=[dusty_tx["tx"]])
+
+        # Sweep tx that doesn't sweep dust ok too
+        sweep_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=[dusty_tx["new_utxos"][0]], version=3)
+        self.nodes[0].rpc.sendrawtransaction(sweep_tx["hex"])
+
+        assert_mempool_contents(self, self.nodes[0], expected=[dusty_tx["tx"], sweep_tx["tx"]])
+
+        self.restart_node(0, extra_args=[])
+        self.restart_node(1, extra_args=[])
+        self.connect_nodes(0, 1)
+
+        assert_equal(self.nodes[0].getrawmempool(), [])
```

Comment:
Thoughts on adding this test before the changes in commit 1?

Reply:
good idea, moved this subtest to `mempool_dust.py` as its own commit, switched it to testing dustrelay arg instead of standardness as I think that's more directly testing what we want.